go/tools/microbench/results.go

```diff
@@ -115,16 +116,51 @@ func MergeMicroBenchmarkDetails(currentMbd, lastReleaseMbd MicroBenchmarkDetails
 	return compareMbs
 }
 
-// ReduceSimpleMedian reduces a MicroBenchmarkDetailsArray by merging
+// ReduceSimpleMedianByName reduces a MicroBenchmarkDetailsArray by merging
 // all MicroBenchmarkDetails with the same benchmark name into a single
 // one. The results of each MicroBenchmarkDetails correspond to the median
 // of the merged elements.
-func (mbd MicroBenchmarkDetailsArray) ReduceSimpleMedian() (reduceMbd MicroBenchmarkDetailsArray) {
+func (mbd MicroBenchmarkDetailsArray) ReduceSimpleMedianByName() (reduceMbd MicroBenchmarkDetailsArray) {
 	sort.SliceStable(mbd, func(i, j int) bool {
-		return mbd[i].Name < mbd[j].Name
+		return mbd[i].PkgName < mbd[j].PkgName && mbd[i].Name < mbd[j].Name
 	})
+	for i := 0; i < len(mbd); {
+		var j int
+		var interOps []int
+		var interNSPerOp []float64
+		var interMBPerSec []float64
+		var interBytesPerOp []float64
+		var interAllocsPerOp []float64
+		for j = i; j < len(mbd) && mbd[i].Name == mbd[j].Name; j++ {
+			interOps = append(interOps, mbd[j].Result.Ops)
+			interNSPerOp = append(interNSPerOp, mbd[j].Result.NSPerOp)
+			interMBPerSec = append(interMBPerSec, mbd[j].Result.MBPerSec)
+			interBytesPerOp = append(interBytesPerOp, mbd[j].Result.BytesPerOp)
+			interAllocsPerOp = append(interAllocsPerOp, mbd[j].Result.AllocsPerOp)
+		}
+
+		interOpsResult := int(math.MedianInt(interOps))
+		interNSPerOpResult := math.MedianFloat(interNSPerOp)
+		interMBPerSecResult := math.MedianFloat(interMBPerSec)
+		interBytesPerOpResult := math.MedianFloat(interBytesPerOp)
+		interAllocsPerOpResult := math.MedianFloat(interAllocsPerOp)
+		reduceMbd = append(reduceMbd, *NewMicroBenchmarkDetails(
+			*NewBenchmarkId(mbd[i].PkgName, mbd[i].Name),
+			mbd[i].GitRef,
+			*NewMicroBenchmarkResult(interOpsResult, interNSPerOpResult, interMBPerSecResult, interBytesPerOpResult, interAllocsPerOpResult),
+		))
+		i = j
+	}
+	return reduceMbd
+}
+
+// ReduceSimpleMedianByGitRef reduces a MicroBenchmarkDetailsArray by merging
+// all MicroBenchmarkDetails with the same git ref into a single
+// one. The results of each MicroBenchmarkDetails correspond to the median
+// of the merged elements.
+func (mbd MicroBenchmarkDetailsArray) ReduceSimpleMedianByGitRef() (reduceMbd MicroBenchmarkDetailsArray) {
```

Comment:
The majority of this function's code is identical to `ReduceSimpleMedianByName`, we can probably do the creation of `inter...` slices, the `append`, and the compute of medians in separate functions that we'll call from both `ReduceSimpleMedianByName` and `ReduceSimpleMedianByGitRef`